enterprise/backup/src/main/java/org/neo4j/backup/impl/OnlineBackupCommandBuilder.java

```diff
@@ -0,0 +1,358 @@
+                .orElse( "" );
+    }
+
+    private String argTimeout()
+    {
+        return Optional.ofNullable( this.timeout )
+                .map( value -> String.format("--timeout=%dms", value) )
+                .orElse( "" );
+    }
+
+    private String argCcOwners()
+    {
+        return Optional.ofNullable( this.consistencyCheckOwners )
+                .map( value -> String.format( "--check-consistency=%b", this.consistencyCheckOwners ) )
+                .orElse( "" );
+    }
+
+    private String argCcLabel()
+    {
+        return Optional.ofNullable( this.consistencyCheckLabel )
+                .map( value -> String.format( "--cc-label-scan-store=%b", this.consistencyCheckLabel ) )
+                .orElse( "" );
+    }
+
+    private String argCcIndexes()
+    {
+        return Optional.ofNullable( this.consistencyCheckIndexes )
+                .map( value -> String.format( "--cc-indexes=%b", this.consistencyCheckIndexes ) )
+                .orElse( "" );
+    }
+
+    private String argCcGraph()
+    {
+        return Optional.ofNullable( this.consistencyCheckGraph )
+                .map( value -> String.format( "--cc-graph=%b", this.consistencyCheckGraph ) )
+                .orElse( "" );
+    }
+
+    private String argAdditionalConf( File backupTarget )
+    {
+        if ( additionalConfig == null )
+        {
+            return "";
+        }
+        File configFile = neo4jHomeFromTarget( backupTarget ).resolve( "../additional_neo4j.conf" ).toFile();
+        writeConfigToFile( additionalConfig, configFile );
+
+        return String.format( "--additional-config=%s", configFile );
+    }
+
+    private void writeConfigToFile( Config config, File file )
+    {
+        try
+        {
+            Writer fileWriter = new BufferedWriter( new FileWriter( file ) );
+            for ( Map.Entry<String,String> entry : config.getRaw().entrySet() )
+            {
+                fileWriter.write( String.format( "%s=%s\n", entry.getKey(), entry.getValue() ) );
+            }
+            fileWriter.close();
+        }
+        catch ( IOException e )
+        {
+            e.printStackTrace();
+        }
+    }
+
+    private String argReportDir()
+    {
+        return Optional.ofNullable( this.consistencyReportLocation )
+                .map( value -> String.format( "--cc-report-dir=%s", value ))
+                .orElse( "" );
+    }
+
+    private String argCheckConsistency()
+    {
+        return Optional.ofNullable( this.checkConsistency )
+                .map( value -> String.format( "--check-consistency=%s", value ) )
+                .orElse( "" );
+    }
+
+    /**
+     * Removes empty args and is a convenience method
+     * @param args nullable, can be empty
+     * @return cleaned command line parameters
+     */
+    private static String[] args( String... args )
+    {
+        List<String> cleanedArgs = Stream.of( args )
+                .filter( Objects::nonNull )
```

Comment:
this whole thing can be replaced with 
`Stream.of( args )
                .filter( StringUtils::isNotEmpty )
                .toArray( String[]::new )` right?

Reply:
Yes, thanks!